### PR TITLE
#158502253-Adds request related end-points

### PR DESCRIPTION
**Pull request by kwanj-k**
### What does this PR do?
Adds ride requests related endpoints

### Description of Task to be completed
* create end-point to create a ride request
* create end-point to get all ride requests
* create end-point to delete/cancel a ride request

### How should I test it?
 1. Clone the repo
  ```
  $ git clone https://github.com/kwanj-k/ride-my-way-API-V1.git
  $ cd into the cloned repo
  $ git checkout ft-requets-#158502253
  ```
Create a virtualenv and activate it::

    Virtualenv -p python3 .
    source bin/activate

Install the dependencies::

    pip install -r requirements.txt 

### Running tests without coverage
Run the tests from the terminal
```
python manage.py test
```

### Running tests with coverage
Run tests with coverage by running this command in the terminal
```
pytest --cov-report term-missing --cov=app
```
## Testing on Postman
Run the development server:
  ```
  $ flask run
  ```
Start Postman

* Make a `POST` request to `/api/v1/rides/1/requests`
* Make a `GET` request to `/api/v1/requests`
* Make a `DELETE` request to `/api/v1/requests/id`

All the end-points should respond accordingly

### What PT stories?
[#158502253]https://www.pivotaltracker.com/story/show/158502253

[#158557140]https://www.pivotaltracker.com/story/show/158557140

[#158576365]https://www.pivotaltracker.com/story/show/158576365

### CODE REVIEW

**Comment by Quantum-35** [comment]
I Like the way you have implemented the models

File: tests/test_ride.py [hunk 1]
```diff
@@ -51,12 +48,12 @@ def test_can_get_ride_by_id(self):
         self.assertEqual(res.status_code, 200)
         
 
-    def test_can_make_ride_request(self):
-        res = self.app.post(
-            '/api/v1/rides/{}'.format(self.data2['id']),
-            data=json.dumps(self.data2),
+    def test_make_ride_request(self):
```
Comment (JoshuaOndieki) [inline tests/test_ride.py]: `test_can_make_ride_request` seemed like a better test name compared to `test_make_ride_request`
Since you'll be testing other edge cases about making a ride request, it could be nice having each test to be precise.

**Comment by kwanj-k** [comment]
Thanks,for the the review,will work on that first

**Comment by TheSteelGuy** [comment]
How many stories is this  pr implementing?

**Comment by TheSteelGuy** [comment]
i guess one with more than a single feature, wise way of reducing stories

**Comment by TheSteelGuy** [comment]
OOh and one more thing i see python complaining that you git complain that you did not append an extra line at the end of some of your files.
